### PR TITLE
add STIR spect matrix

### PR DESCRIPTION
**Pull request by KrisThielemans**
This was easy to do and allows using SPECT just like SPECT.

### CODE REVIEW

**Comment by KrisThielemans** [comment]
This now needs https://github.com/UCL/STIR/pull/322

**Comment by KrisThielemans** [comment]
needs a merge with master and checking conflicts in CHANGES.md

File: src/xSTIR/pSTIR/STIR.py [hunk 1]
```diff
@@ -750,6 +750,68 @@ def get_num_tangential_LORs(self):
...
+        '''
+        Returns a bool checking if we're keeping the whole matrix in memory or not.
+        '''
+        return _int_par(self.handle, self.name, 'keep_all_views_in_cache') != 0
+    def set_attenuation_image(self, value):
+        '''
+        Sets the attenuation image used by the projector.
+        '''
+        assert_validity(value, ImageData)
+        _setParameter(self.handle, self.name, 'attenuation_image', value.handle)
```
Comment (samdporter) [inline src/xSTIR/pSTIR/STIR.py:790]: probably has to be
```
parms.set_parameter(self.handle, self.name, 'attenuation_image', value.handle)
```

File: src/xSTIR/pSTIR/STIR.py [hunk 2]
```diff
@@ -750,6 +750,68 @@ def get_num_tangential_LORs(self):
...
+            pyiutil.deleteDataHandle(self.handle)
+    def set_keep_all_views_in_cache(self, value):
+        '''
+        Enable keeping the matrix in memory.
+
+        This speeds-up the calculations, but can use a lot of memory.
+
+        You have to call set_up() after this (unless the value didn't change).
+        '''
+        _set_int_par(self.handle, self.name, 'keep_all_views_in_cache', value)
```
Comment (samdporter) [inline src/xSTIR/pSTIR/STIR.py:778]: problem with _set_in_par 
not defined

File: src/xSTIR/pSTIR/STIR.py [hunk 3]
```diff
@@ -750,6 +750,68 @@ def get_num_tangential_LORs(self):
...
+        '''
+        assert_validity(value, ImageData)
+        _setParameter(self.handle, self.name, 'attenuation_image', value.handle)
+        return self
+    def get_attenuation_image(self):
+        '''
+        Returns the attenuation image used by the projector.
+        '''
+        image = ImageData()
+        image.handle = _getParameterHandle(self.handle, self.name, 'attenuation_image')
```
Comment (samdporter) [inline src/xSTIR/pSTIR/STIR.py:797]: problem with _getParameterHandle
not defined

File: examples/Python/SPECT/acquisition_model.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    show_2D_array('Phantom image', image_array[z,:,:])
+
+    # select acquisition model that implements the geometric
+    # forward projection by a ray tracing matrix multiplication
+    acq_model_matrix = SPECTUBMatrix();
+    acq_model = AcquisitionModelUsingMatrix(acq_model_matrix)
+
+    print('projecting image...')
+    # project the image to obtain simulated acquisition data
+    # data from raw_data_file is used as a template
```
Comment (samdporter) [inline examples/Python/SPECT/acquisition_model.py:107]: needs `image = image.zoom_image(zooms=(0.5, 1.0, 1.0), size=(12, -1, -1))` in order to work

File: examples/Python/SPECT/acquisition_model.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+You probably want to check that instead.
+'''
+
+## CCP PETMR Synergistic Image Reconstruction Framework (SIRF)
+## Copyright 2015 - 2017 Rutherford Appleton Laboratory STFC
+## Copyright 2015 - 2019 University College London.
+##
+## This is software developed for the Collaborative Computational
+## Project in Positron Emission Tomography and Magnetic Resonance imaging
+## (http://www.ccppetmr.ac.uk/).
```
Comment (KrisThielemans) [inline examples/Python/SPECT/acquisition_model.py:23]: Need to update to SyneRBI

File: examples/Python/SPECT/interactive/basic_reconstruction.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+# This demo is a 'script', i.e. intended to be run step by step in a
+# Python IDE such as spyder. It is organised in 'cells'. spyder displays these
+# cells nicely and allows you to run each cell on its own.
+#
+# WARNING: This script assumes you have run the display_and_projection.py demo first!
+#
+# Author: Kris Thielemans
+# Author: Evgueni Ovtchinnikov
+#
+## CCP PETMR Synergistic Image Reconstruction Framework (SIRF)
```
Comment (KrisThielemans) [inline examples/Python/SPECT/interactive/basic_reconstruction.py:15]: synerbi

File: examples/Python/SPECT/interactive/display_and_projection.py [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+# anything depending on your IDE settings (check the 'backend' settings).
+# For instance, in spyder, go to Tools->Preferences->iPython->Graphics and
+# set your backend to "automatic". You will have to do this BEFORE you start the
+# ipython console (or just restart spyder)
+#
+# Author: Kris Thielemans
+# First version: 8th of September 2016
+#
+
+## CCP PETMR Synergistic Image Reconstruction Framework (SIRF)
```
Comment (KrisThielemans) [inline examples/Python/SPECT/interactive/display_and_projection.py:22]: synerbi

**Comment by KrisThielemans** [comment]
@evgueni-ovtchinnikov this should be ready to merge now. @samdporter made some fixes. I don't think there are any tests yet, so we'll have to flag this as preliminary in the CHANGES.md. We don't have anything about it in the User's Guide either, but I'll try to add a few minimal lines (or possibly @samdporter could...)

**Pull request merged**